examples/keras_io/nlp/neural_machine_translation_with_keras_nlp.py

```diff
@@ -0,0 +1,515 @@
+to True. This will then be propagated to all subsequent layers.
+"""
+
+# Encoder
+encoder_inputs = keras.Input(
+    shape=(None,), dtype="int64", name="encoder_inputs"
+)
+
+x = keras_nlp.layers.TokenAndPositionEmbedding(
+    vocabulary_size=ENG_VOCAB_SIZE,
+    sequence_length=MAX_SEQUENCE_LENGTH,
+    embedding_dim=EMBED_DIM,
+)(encoder_inputs)
+
+encoder_outputs = keras_nlp.layers.TransformerEncoder(
+    intermediate_dim=INTERMEDIATE_DIM, num_heads=NUM_HEADS
+)(inputs=x)
+encoder = keras.Model(encoder_inputs, encoder_outputs)
+
+
+# Decoder
+decoder_inputs = keras.Input(
+    shape=(None,), dtype="int64", name="decoder_inputs"
+)
+encoded_seq_inputs = keras.Input(
+    shape=(None, EMBED_DIM), name="decoder_state_inputs"
+)
+
+x = keras_nlp.layers.TokenAndPositionEmbedding(
+    vocabulary_size=SPA_VOCAB_SIZE,
+    sequence_length=MAX_SEQUENCE_LENGTH,
+    embedding_dim=EMBED_DIM,
+)(decoder_inputs)
+
+x = keras_nlp.layers.TransformerDecoder(
+    intermediate_dim=INTERMEDIATE_DIM, num_heads=NUM_HEADS
+)(decoder_sequence=x, encoder_sequence=encoded_seq_inputs)
+x = keras.layers.Dropout(0.5)(x)
+decoder_outputs = keras.layers.Dense(SPA_VOCAB_SIZE, activation="softmax")(x)
+decoder = keras.Model(
+    [
+        decoder_inputs,
+        encoded_seq_inputs,
+    ],
+    decoder_outputs,
+)
+decoder_outputs = decoder([decoder_inputs, encoder_outputs])
+
+transformer = keras.Model(
+    [encoder_inputs, decoder_inputs],
+    decoder_outputs,
+    name="transformer",
+)
+
+"""
+## Training our model
+
+We'll use accuracy as a quick way to monitor training progress on the validation data.
+Note that machine translation typically uses BLEU scores as well as other metrics,
+rather than accuracy. However, in order to use metrics like ROUGE, BLEU, etc. we
+will have decode the probabilities and generate the text. Text generation is
+computationally expensive, and performing this during training is not recommended.
+
+Here we only train for 1 epoch, but to get the model to actually converge
+you should train for at least 10 epochs.
+"""
+
+transformer.summary()
+transformer.compile(
+    "rmsprop", loss="sparse_categorical_crossentropy", metrics=["accuracy"]
+)
+transformer.fit(train_ds, epochs=EPOCHS, validation_data=val_ds)
+
+"""
+## Decoding test sentences (qualitative analysis)
+
+Finally, let's demonstrate how to translate brand new English sentences.
+We simply feed into the model the tokenized English sentence
+as well as the target token `"[START]"`. The model outputs probabilities of the
+next token. We then we repeatedly generated the next token conditioned on the
+tokens generated so far, until we hit the token `"[END]"`.
+
+For decoding, we will use the `keras_nlp.samplers` module from
+KerasNLP. Greedy Decoding is a text decoding method which outputs the most
+likely next token at each time step, i.e., the token with the highest probability.
+"""
+
+
+def decode_sequences(input_sentences):
+    batch_size = 1
```

Comment:
Why the change compared to the more general `ops.shape(input_sentences)[0]`?

Reply:
Because I change the what were passed to `decode_sequences()`, `input_sentences` is not a tensor.
Before, the `decode_sequences()`, takes a TF tensor (`decode_sequences(tf.constant([input_sentence]))`), but it seems Torch doesn't have string tensor, I change `decode_sequences()`  to take a list of string.

Reply:
Got it.